File: src/main/java/me/rsman/BetterMinecraftCore/Managers/Command/CommandCompletionsManager.java
package me.rsman.BetterMinecraftCore.Managers.Command;

import co.aikar.commands.BukkitCommandCompletionContext;
import co.aikar.commands.CommandCompletions;
import me.rsman.BetterMinecraftCore.Managers.CraftManager;
import me.rsman.BetterMinecraftCore.Managers.EnchantManager;
import me.rsman.BetterMinecraftCore.Managers.ItemManager;
import me.rsman.BetterMinecraftCore.configs.containers.BmcCraftContainer;
import me.rsman.BetterMinecraftCore.configs.containers.BmcItemContainer;
import me.rsman.BetterMinecraftCore.enums.EAttributes;
import me.rsman.BetterMinecraftCore.enums.EEnchants;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.Set;
import java.util.stream.Collectors;

public final class CommandCompletionsManager {

    public static void init(){
        registerCommandCompletions();
    }

    private static void registerCommandCompletions() {

        CommandCompletions<BukkitCommandCompletionContext> commandCompletions = CommandManager.get().getCommandCompletions();
        commandCompletions.registerAsyncCompletion("item", c -> {
            CommandSender sender = c.getSender();
            if (sender instanceof Player) {
                Set<String> itemList = BmcItemContainer.getInstance().getItems().keySet();
                return new ArrayList<>(itemList);
            }
            return null;
        });
        commandCompletions.registerAsyncCompletion("attribute", c -> {
            CommandSender sender = c.getSender();
            if (sender instanceof Player) {
                return EAttributes.getAllKeys();
            }
            return null;
        });
        commandCompletions.registerAsyncCompletion("enchantment", c -> {
            CommandSender sender = c.getSender();
            if (sender instanceof Player) {
                return EEnchants.getNonReplacedEnumKeys();
            }
            return null;
        });
        commandCompletions.registerAsyncCompletion("shapedCrafts", c -> {
            CommandSender sender = c.getSender();
            if (sender instanceof Player) {
                return BmcCraftContainer.getShapedKeys();
            }
            return null;
        });
        commandCompletions.registerAsyncCompletion("shapelessCrafts", c -> {
            CommandSender sender = c.getSender();
            if (sender instanceof Player) {
                return BmcCraftContainer.getShapelessKeys();
            }
            return null;
        });
    }
}

File: src/main/kotlin/me/rsman/BetterMinecraftCore/Managers/Command/CommandCompletionsManager.kt
package me.rsman.BetterMinecraftCore.Managers.Command

import me.rsman.BetterMinecraftCore.configs.containers.BmcCraftContainer.Companion.getShapedKeys
import me.rsman.BetterMinecraftCore.configs.containers.BmcCraftContainer.Companion.getShapelessKeys
import me.rsman.BetterMinecraftCore.enums.EEnchants
import me.rsman.BetterMinecraftCore.enums.EAttributes
import co.aikar.commands.CommandCompletions
import co.aikar.commands.BukkitCommandCompletionContext
import org.bukkit.entity.Player
import me.rsman.BetterMinecraftCore.configs.containers.BmcItemContainer
import java.util.ArrayList

object CommandCompletionsManager {
    fun init() {
        registerCommandCompletions()
    }

    private fun registerCommandCompletions() {
        val commandCompletions: CommandCompletions<BukkitCommandCompletionContext> = CommandManager.get()!!.getCommandCompletions()
        commandCompletions.registerAsyncCompletion("item") { c: BukkitCommandCompletionContext ->
            val sender = c.sender
            if (sender is Player) {
                val itemList: Set<String> = BmcItemContainer.instance!!.items!!.keys
                return@registerAsyncCompletion ArrayList(itemList)
            }
            null
        }
        commandCompletions.registerAsyncCompletion("itemOrVanilla") { c: BukkitCommandCompletionContext ->
            val sender = c.sender
            if (sender is Player) {
                val itemList: Set<String> = setOf("<vanilla_item_ex:_m.ITEM_ID>") + BmcItemContainer.instance!!.items!!.keys
                return@registerAsyncCompletion ArrayList(itemList)
            }
            null
        }
        commandCompletions.registerAsyncCompletion("attribute") { c: BukkitCommandCompletionContext ->
            val sender = c.sender
            if (sender is Player) {
                return@registerAsyncCompletion EAttributes.Companion.allKeys
            }
            null
        }
        commandCompletions.registerAsyncCompletion("enchantment") { c: BukkitCommandCompletionContext ->
            val sender = c.sender
            if (sender is Player) {
                return@registerAsyncCompletion EEnchants.Companion.nonReplacedEnumKeys
            }
            null
        }
        commandCompletions.registerAsyncCompletion("shapedCrafts") { c: BukkitCommandCompletionContext ->
            val sender = c.sender
            if (sender is Player) {
                return@registerAsyncCompletion getShapedKeys()
            }
            null
        }
        commandCompletions.registerAsyncCompletion("shapelessCrafts") { c: BukkitCommandCompletionContext ->
            val sender = c.sender
            if (sender is Player) {
                return@registerAsyncCompletion getShapelessKeys()
            }
            null
        }
    }
}